src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringEndsWithTranslator.cs

```diff
@@ -19,35 +20,45 @@
 // AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
 // ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
 // TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
 #endregion
 
 using System.Linq.Expressions;
 using System.Reflection;
+using JetBrains.Annotations;
 using Microsoft.EntityFrameworkCore.Query.Expressions;
 using Microsoft.EntityFrameworkCore.Query.ExpressionTranslators;
 
 namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal
 {
+    /// <summary>
+    /// Translates <see cref="M:string.EndsWith(string)"/> to 'RIGHT(text, int) = text'.
+    /// </summary>
     public class NpgsqlStringEndsWithTranslator : IMethodCallTranslator
     {
-        static readonly MethodInfo _methodInfo
-            = typeof(string).GetRuntimeMethod(nameof(string.EndsWith), new[] { typeof(string) });
+        static readonly MethodInfo MethodInfo =
+            typeof(string).GetRuntimeMethod(nameof(string.EndsWith), new[] { typeof(string) });
+
+        /// <inheritdoc />
+        [CanBeNull]
+        public virtual Expression Translate(MethodCallExpression e)
```

Comment:
Isn't the previous way with an expression-bodied method better? I don't really see a reason for this change, just to extra the `instance` variable?

Reply:
We should decline to translate when the `instance` is set to `null`. This shouldn't be possible directly from user code (e.g. should at least be `Expression.Constant(null)`). But a previously executed translator or plugin could potentially set it to `null`, and I would rather pay the check here than risk an unexpected translation.

Reply:
This is just a styling note: I was suggesting to write this as an expression-bodied method, testing whether `e.Object` is null. Yes, it means we write `e.Object` several times instead of `instance`, but the two are literally the same number of characters :)

Reply:
Oh! Sorry, totally misunderstood. Yes, I can rewrite as such.